apps/telegram_bot/main.py

```diff
@@ -0,0 +1,348 @@
+    PlayHtSynthesizer,
+    AzureSynthesizer,
+    CoquiTTSSynthesizer,
+    RimeSynthesizer,
+]
+""" CONFIGURATION START """
+# Required environment variables containing API key: OPENAI_API_KEY, TELEGRAM_BOT_KEY, and your Vocode synthesizers classes corresponding API key variable
+
+# Your chosen synthesizer provider's corresponding Vocode turn_based class instance
+SYNTH = CoquiSynthesizer(use_xtts=True)
+
+# Array of tuples (synthesizer's voice id, nickname, description if text to voice)
+DEFAULT_VOICES = [(None, "Coqui Default", None)]
+
+# Instructions to the language model responsible for generating response transcript.
+SYSTEM_PROMPT = "You are having a voice-to-voice conversation with a user. Always pretend you are a real human. Do not avoid or decline to answer questions because you are not human."
+""" CONFIGURATION END """
+
+# key=synth class, value=attribute that chooses the provider voice
+voice_attr_of: Dict[Type[BaseSynthesizer], str] = {
+    ElevenLabsSynthesizer: "voice_id",
+    PlayHtSynthesizer: "voice",
+    StreamElementsSynthesizer: "voice",
+    AzureSynthesizer: "voice_name",
+    CoquiSynthesizer: "voice_id",
+    CoquiTTSSynthesizer: "speaker",
+    RimeSynthesizer: "speaker",
+}
+assert set(voice_attr_of.keys()) == set(
+    supported_classes
+), "supported_classes must match the keys of voice_attr_of!"
+
+assert (
+    type(SYNTH) in voice_attr_of.keys()
+), "Synthesizer class must be one of the supported ones!"
+# Check voice_attr_of is correct by asserting all classes have their corresponding value as a parameter in the init function
+for key, value in voice_attr_of.items():
+    assert value in inspect.signature(key.__init__).parameters
+
+
+# Special defaultdict that supports replitdb
+class defaultdict:
+    def __init__(self, db) -> None:
+        # Initialize an empty dictionary to store user data
+        self.db = db
+
+    def __getitem__(self, chat_id) -> Dict:
+        # Return the user data for a given user id, or create a new one if not found
+        if chat_id not in self.db:
+            self.db[chat_id] = {
+                "voices": DEFAULT_VOICES,
+                "current_voice": DEFAULT_VOICES[0],
+                "current_conversation": None,
+            }
+        return self.db[chat_id]
+
+    def __setitem__(self, chat_id, user_data) -> None:
+        # Set the user data for a given user id
+        self.db[chat_id] = user_data
+
+
+class VocodeBotResponder:
+    def __init__(
+        self,
+        transcriber: BaseTranscriber,
+        system_prompt: str,
+        synthesizer: BaseSynthesizer,
+        db=None,
+    ) -> None:
+        self.transcriber = transcriber
+        self.system_prompt = system_prompt
+        self.synthesizer = synthesizer
+        self.db = defaultdict(db if db else {})
+
+    def get_agent(self, chat_id: int) -> ChatGPTAgent:
+        # Get current voice name and description from DB
+        _, voice_name, voice_description = self.db[chat_id].get(
```

Comment:
nit: turn this `get` call into a helper method on `ChatsDB` since we use it in `get_response`

Reply:
replaced ChatsDB with defaultdict